containers/Dockerfile.buildenv

```diff
@@ -304,19 +247,16 @@ RUN git clone --single-branch --branch v7.0.0 --depth 1 \
     && cd /work/charm_7_0_0 \
     && git checkout v7.0.0 \
     && git apply /work/v7.0.0.patch \
-    && ./build LIBS multicore-linux-${CHARM_ARCH} gcc \
-      ${PARALLEL_MAKE_ARG} -g -O2 --build-shared \
+    && ./build LIBS multicore-linux-${CHARM_ARCH} clang \
```

Comment:
sorry why switch to clang?

Reply:
I may have had a reason but don't remember, maybe just consistency with the MPI Charm build below. Prefer to drop this change?

Reply:
I think I intentionally wanted different builds just to make sure we could do both GCC & clang with charm++ :shrug: So slight preference to drop

Reply:
ok, dropped 👍